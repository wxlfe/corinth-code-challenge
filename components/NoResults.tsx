const NoResults = () => {

    const options = [
        'If an item does not appear in our records, it does not exist.',
        'These are not the results you\'re looking for.',
        'Perhaps the archives are incomplete.',
        'I\'ve got a bad feeling about this.'
    ];

    return (
        <>
            <h3 className='no-results-header'>No Results Found</h3>
            <p className='no-results-body'>
                { options[Math.floor(Math.random() * 4)] }
            </p>
        </>
    )
}

export default NoResults