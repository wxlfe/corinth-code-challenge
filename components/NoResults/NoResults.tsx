import { Text } from '@nextui-org/react';

type Props = {
  overrideOption?: number;
};

const NoResults = ({ overrideOption }: Props) => {
  const options = [
    'If an item does not appear in our records, it does not exist.',
    "These are not the results you're looking for.",
    'Perhaps the archives are incomplete.',
    "I've got a bad feeling about this.",
  ];

  return (
    <>
      <Text
        h3
        color='primary'
        css={{ textAlign: 'center', paddingTop: '1rem' }}
      >
        No Results Found
      </Text>
      <Text
        color='primary'
        css={{ textAlign: 'center', paddingBottom: '1rem' }}
      >
        {
          options[
            !overrideOption ? Math.floor(Math.random() * 4) : overrideOption
          ]
        }
      </Text>
    </>
  );
};

export default NoResults;
