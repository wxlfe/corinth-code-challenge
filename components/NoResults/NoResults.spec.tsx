import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import NoResults from './NoResults';

describe('No Results Component', () => {
  test('should create', () => {
    expect(NoResults).toBeTruthy();
  });

  test('should render', () => {
    const { asFragment } = render(<NoResults overrideOption={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should say No Results Found', () => {
    const { asFragment } = render(<NoResults />);
    expect(asFragment()).toHaveTextContent('No Results Found');
  });
});
