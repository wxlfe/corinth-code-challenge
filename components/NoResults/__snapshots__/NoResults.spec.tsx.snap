// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Results Component should render 1`] = `
<DocumentFragment>
  <h3
    class="nextui-c-PJLV nextui-c-PJLV-isiIAY-css"
  >
    No Results Found
  </h3>
  <p
    class="nextui-c-PJLV nextui-c-PJLV-ifVnRkg-css"
  >
    These are not the results you're looking for.
  </p>
</DocumentFragment>
`;
